api/v1alpha1/observatorium_types.go

```diff
@@ -378,6 +378,17 @@ type LokiSpec struct {
 	Version string `json:"version,omitempty"`
 	// VolumeClaimTemplate
 	VolumeClaimTemplate VolumeClaimTemplate `json:"volumeClaimTemplate"`
+	// Compute Resources required by each component containers.
+	// +optional
+	Resources *LokiResourcesSpec `json:"resources,omitempty"`
+}
+
+type LokiResourcesSpec struct {
+	Compactor *v1.ResourceRequirements `json:"compactor,omitempty"`
+	Distributor *v1.ResourceRequirements `json:"distributor,omitempty"`
+	Ingester *v1.ResourceRequirements `json:"ingester,omitempty"`
+	Querier *v1.ResourceRequirements `json:"querier,omitempty"`
+	QueryFrontend *v1.ResourceRequirements `json:"query_frontend,omitempty"`
```

Comment:
As mentioned on the [observatorium/observatorium#411](https://github.com/observatorium/observatorium/pull/411#discussion_r642223308) PR, this needs to slightly change to:
```go
type LokiSpec struct {
    Components *LokiComponentsSpec
}

type LokiComponentsSpec struct {
    Compactor *LokiComponentSpec
    ...
}
type LokiComponentSpec {
   Resources *v1.ResourceRequirements `json:"resources,omitempty"`
}
```